src/agent/protocols/hack/update-generated-proto.sh

```diff
@@ -1,68 +1,133 @@
 #!/bin/bash
 
 die() {
+    echo "===================================================================="
+    echo "====                compile protocols failed                    ===="
     echo $1
-    exit
+    echo "===================================================================="
+    exit 1
 }
 
-get_source_version() {
-    if [ ! -d $GOPATH/src/$1 ]; then
-        go get -d -v $1
-    fi
-    [ $? -eq 0 ] || die "Failed to get $1"
-    if [ "$2" != "" ] ; then
-	    pushd "${GOPATH}/src/$1"
-        if [ $(git rev-parse HEAD) != $2 ] ; then
-            git checkout $2
-            [ $? -eq 0 ] || die "Failed to get $1 $2"
-        fi
-        popd
-    fi
+show_succeed_msg() {
+    echo "===================================================================="
+    echo "====                                                            ===="
+    echo "====                compile protocols succeed                   ===="
+    echo "====                                                            ===="
+    echo "===================================================================="
+}
+
+show_usage() {
+    echo "===================================================================="
+    echo "===    USAGE: make PROTO_FILE=XYZ.proto generate-protocols       ==="
+    echo "===                                                              ==="
+    echo "===    where PROTO_FILE may be:                                  ==="
+    echo "===      all: will compile all proto files                       ==="
+    echo "===      agent.proto                                             ==="
```

Comment:
Can you find a way to avoid hard-coding this list of files? Maybe make `golang_files` global (and rename to `proto_files` for clarity) or pass in a string containing the list of proto files.

Reply:
The proto files list are used the `proto_files_list` variable now.